src/main/shell-session/shell-env-modifier/shell-env-modifiers.ts

```diff
@@ -0,0 +1,18 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+
+import { computed, IComputedValue } from "mobx";
+import type { LensMainExtension } from "../../../extensions/lens-main-extension";
+ 
+interface Dependencies {
+  extensions: IComputedValue<LensMainExtension[]>;
+}
+ 
+export const shellEnvModifiers = ({ extensions }: Dependencies) => {
+  return computed(() => (
+    extensions.get()
+      .flatMap((extension) => extension.shellEnvModifier)
```

Comment:
And thus a `.filter(Boolean)` here too. Also this can be just `.map` because `shellEnvModifier` is not an array.